tests/util.test.ts

```diff
@@ -56,6 +56,85 @@ describe('Zip', function () {
   });
 });
 
+describe('Parse KV pairs', function () {
+  it('parse single unquoted envVar', async function () {
+    const envVar = 'KEY1=VALUE1';
+    const pairs = parseKVPairs(envVar);
+    expect(pairs).deep.equal({ KEY1: 'VALUE1' });
+  });
+  it('parse single quoted envVar', async function () {
+    const envVar = 'KEY1="VALUE1"';
+    const pairs = parseKVPairs(envVar);
+    expect(pairs).deep.equal({ KEY1: 'VALUE1' });
+  });
+  it('parse multiple unquoted envVars', async function () {
+    const envVars = 'KEY1=VALUE1,KEY2=VALUE2';
+    const pairs = parseKVPairs(envVars);
+    expect(pairs).deep.equal({ KEY1: 'VALUE1', KEY2: 'VALUE2' });
+  });
+  it('parse multiple quoted envVars', async function () {
+    const envVars = 'KEY1="VALUE1",KEY2="VALUE2"';
+    const pairs = parseKVPairs(envVars);
+    expect(pairs).deep.equal({ KEY1: 'VALUE1', KEY2: 'VALUE2' });
+  });
+  it('parse mix of quoted and unquoted envVars', async function () {
+    const envVars = 'KEY1=VALUE1,KEY2="VALUE2"';
+    const pairs = parseKVPairs(envVars);
+    expect(pairs).deep.equal({ KEY1: 'VALUE1', KEY2: 'VALUE2' });
+  });
+  it('parse envVars with multiple = characters', async function () {
+    const envVars = 'KEY1=VALUE=1,KEY2=VALUE=2';
+    const pairs = parseKVPairs(envVars);
+    expect(pairs).deep.equal({ KEY1: 'VALUE=1', KEY2: 'VALUE=2' });
+  });
+  it('throws an error if envVars is malformed', async function () {
+    const envVars = 'KEY1,VALUE1';
+    expect(function () {
+      parseKVPairs(envVars);
+    }).to.throw(
+      'The expected data format should be "KEY1=VALUE1", got "KEY1" while parsing "KEY1,VALUE1"',
+    );
+  });
+  it('throws an error if envVars are not quoted correctly', async function () {
+    const envVars = 'KEY1="VALUE1.1,VALUE1.2,KEY2="VALUE2"';
+    expect(function () {
+      parseKVPairs(envVars);
+    }).to.throw(
+      `The expected data format should be "KEY1=VALUE1", got "VALUE1.2" while parsing "KEY1="VALUE1.1,VALUE1.2,KEY2="VALUE2""`,
+    );
+  });
+});
```

Comment:
Lets use a table test for this which maybe more readable
example: https://github.com/google-github-actions/upload-cloud-storage/blob/11a34288710191ea7fdfeee3f10c7fe4a356d2cd/tests/util.test.ts#L24

Reply:
Done! This should be easier to extend with more cases in future if needed.